client/state/sites/plans/reducer.js

```diff
@@ -1,39 +1,86 @@
 /**
+ * External dependencies
+ */
+import omit from 'lodash/object/omit';
+
+/**
  * Internal dependencies
  */
 import {
-	FETCH_SITE_PLANS,
-	FETCH_SITE_PLANS_COMPLETED,
-	REMOVE_SITE_PLANS
-} from './action-types';
-import omit from 'lodash/object/omit';
+	SITE_PLANS_FETCH,
+	SITE_PLANS_FETCH_COMPLETED,
+	SITE_PLANS_FETCH_FAILED,
+	SITE_PLANS_REMOVE,
+	SITE_PLANS_TRIAL_CANCEL,
+	SITE_PLANS_TRIAL_CANCEL_COMPLETED,
+	SITE_PLANS_TRIAL_CANCEL_FAILED
+} from 'state/action-types';
 
 export const initialSiteState = {
+	data: null,
 	error: null,
 	hasLoadedFromServer: false,
 	isFetching: false,
-	data: null
+	isUpdating: false
```

Comment:
Are there any other places in Calypso where we use three booleans to describe the state of data as it relates to the server? I _think_ every state we need to handle in the sites plans components could be handled with only two:

- One to indicate whether or not the data is valid. That is `hasLoadedFromServer` in this case, which is perhaps not be the best name if we decide to flip it to invalidate local data.
- One to indicate whether or not a request is awaiting a response that might update the data (both `isFetching` and `isUpdating`) right now.

We always use `isUpdating` and `isFetching` separately below, so we might be able to combine them into one property and use `hasLoadedFromServer` (or something with a different name) to invalidate the data in memory.